### DISCUSSION TITLE
PowerDNS and dnsdist Prometheus Metrics Discrepancy

### DISCUSSION THREAD

**Discussion by v0tti** [PowerDNS/pdns#13606] [Q&A] [answered]
Hi everyone,
I've been working with PowerDNS and dnsdist, and I've noticed a significant difference between the `dnsdist_responses` and `pdns_auth_backend_queries` Prometheus metrics. Even though caching in dnsdist is disabled, the rate of dnsdist responses is higher than the rate of PowerDNS queries. I'm curious to understand the reasons behind this discrepancy.

**Comment by Habbie** [comment, marked as answer]
The `backend-queries` metric in the Authoritative server counts queries sent by PowerDNS to the database backend. This number can be higher than the number of DNS queries answered (because we may need multiple pieces of information from the database to answer one query), or lower (because the auth also has caches in several places).

I suspect `pdns_auth_tcp_answers` + `pdns_auth_udp_answers` should come closer to `dnsdist_responses`

**Reply by v0tti** [reply]
Thanks for the explanation. `pdns_auth_tcp_answers` + `pdns_auth_udp_answers` works out to pretty much the same as `dnsdist_responses`.
